src/renderer/component/socialShare/view.jsx

```diff
@@ -0,0 +1,100 @@
+// @flow
+import React from 'react';
+import type { Claim } from 'types/claim';
+import Button from 'component/button';
+import { clipboard } from 'electron';
+import { FormRow } from 'component/common/form';
+import * as icons from 'constants/icons';
+import Tooltip from 'component/common/tooltip';
+
+type Props = {
+  claim: Claim,
+  onDone: () => void,
+};
+
+class SocialShare extends React.PureComponent<Props> {
+  constructor(props: Props) {
+    super(props);
+
+    this.input = undefined;
+  }
+
+  input: ?HTMLInputElement;
+
+  render() {
+    const { claim_id: claimId, name: claimName, channel_name: channelName } = this.props.claim;
+    const { onDone } = this.props;
+    const speechPrefix = 'http://spee.ch/';
+    const speechURL = channelName
+      ? `${speechPrefix}${channelName}/${claimName}`
+      : `${speechPrefix}${claimName}#${claimId}`;
+
+    return (
+      <div>
+        <div className="card__title">
+          <h3>{__('Share')}</h3>
+          <div className="card__content">
+            <FormRow verticallyCentered padded stretch>
+              <input
+                className="input-copyable form-field__input"
+                readOnly
+                value={speechURL || ''}
+                ref={input => {
+                  this.input = input;
+                }}
+                onFocus={() => {
+                  if (this.input) {
+                    this.input.select();
+                  }
+                }}
+              />
+              <Button
+                noPadding
+                button="secondary"
+                icon={icons.CLIPBOARD}
+                onClick={() => {
+                  clipboard.writeText(speechURL);
+                }}
+              />
+            </FormRow>
```

Comment:
Since the above code is pretty much identical to the `Address` component at `component/address/view.jsx` we should probably break this into it's own component.

Maybe `CopyableInput`?

I'm only talking about everything inside of the `FormRow`. It would just need two props (I think), `value` and `copySuccessMessage` (for the snackbar after you copy it)

Reply:
I originally tried to use <Address /> and it crashed at that point. Using <Address /> now has the odd behavior that it doesn't show the snackbar until after the modal is closed (modal over modal?). I will need to really look into how doNotify on top of doNotify works. Perhaps it's putting the next notification in a queue that doesn't get activated until my modal is closed?

If so, taking doNotify functionality out of Address and making it an optional passed prop might be best?

Reply:
Ah you are right. I forgot about that. I just created an issue for it.
https://github.com/lbryio/lbry-desktop/issues/1945

I'm fine with passing in a prop for now to disable this (it should default to notify) for now. Just include a link to this issue #1945 in a comment above it.

We should probably rename the `Address` component if we are going to use it here.